### PR TITLE
feat(dal,si-events): Move node weights off the graph

### PR DESCRIPTION
**Pull request by zacharyhamm**
Moves node weights off the internal petgraph, storing them in the layerdb. Local node weights now only store the address and the merkel tree hash (although we may move that off the local node weight as well).

### CODE REVIEW

**Comment by zacharyhamm** [comment]
bors try

**Comment by zacharyhamm** [comment]
bors try

File: prelude-si/macros/rust.bzl [hunk 1]
```diff
@@ -156,6 +156,39 @@ def rust_binary(
...
+        resources = resources,
+        proc_macro = proc_macro,
+        visibility = visibility,
+        **kwargs
+    )
+
+    _alias(
+        name = "integration_test",
+        actual = ":{}".format(name),
+    )
```
Comment (nickgerace) [inline prelude-si/macros/rust.bzl:190]: This should work great!

File: lib/dal/tests/integration_test/graph/mod.rs
Comment (nickgerace) [inline lib/dal/tests/integration_test/graph/mod.rs:1]: Ah this is the fabled unit tests for the graph become integration tests... looks good to me

File: lib/dal/tests/integration_test/graph/mod.rs [hunk 2]
```diff
@@ -0,0 +1,3880 @@
...
+// async fn new(ctx: DalContext) {
+//     let change_set = ChangeSet::new_local().expect("Unable to create ChangeSet");
+//     let change_set = &change_set;
+//     let graph =
+//         WorkspaceSnapshotGraph::new(change_set).expect("Unable to create WorkspaceSnapshotGraph");
+//     assert!(graph.is_acyclic_directed());
+// }
+
+// Previously, WorkspaceSnapshotGraph::new would not populate its node_index_by_id, so this test
+// would fail, in addition to any functionality that depended on getting the root node index
```
Comment (nickgerace) [inline lib/dal/tests/integration_test/graph/mod.rs:45]: Got it

File: lib/dal/src/component.rs [hunk 3]
```diff
@@ -1242,9 +1243,6 @@ impl Component {
             ctx.workspace_snapshot()?
                 .add_node(NodeWeight::Component(new_component_node_weight))
                 .await?;
-            ctx.workspace_snapshot()?
-                .replace_references(component_idx)
-                .await?;
```
Comment (nickgerace) [inline lib/dal/src/component.rs:1247]: Do we no longer need this?

**Review by nickgerace — APPROVED** [review]

File: lib/dal/src/workspace_snapshot/graph.rs [hunk 4]
```diff
@@ -875,390 +496,66 @@ impl WorkspaceSnapshotGraph {
...
+        address: NodeWeightAddress,
+    ) -> WorkspaceSnapshotGraphResult<Ulid> {
+        match self.id_by_node_address.get(&address).copied().ok_or(
+            WorkspaceSnapshotGraphError::NodeWithAddressNotFound(address),
+        ) {
+            Ok(a) => Ok(a),
+            Err(e) => {
+                error!("{}", e);
+                Err(e)
+            }
```
Comment (nickgerace) [inline lib/dal/src/workspace_snapshot/graph.rs:533]: I presume we want to keep this logging for now?

File: lib/dal/src/component.rs (hunk 3 above)
Comment (zacharyhamm) [reply lib/dal/src/component.rs:1247]: add_node decides if it needs to replace refs now

**Pull request merged**